Author: Danny Auble <da@schedmd.com>
Date: Mon, 22 Jan 2018 13:22:46 -0700
Description: Fix issues when starting the backup slurmdbd.

Commit: 8bb58a3116e067817c9eea6834d293322667eb3c
Upstream-Bug: 4656

Index: slurm-llnl/src/plugins/accounting_storage/mysql/accounting_storage_mysql.c
===================================================================
--- slurm-llnl.orig/src/plugins/accounting_storage/mysql/accounting_storage_mysql.c
+++ slurm-llnl/src/plugins/accounting_storage/mysql/accounting_storage_mysql.c
@@ -843,19 +843,12 @@ static int _as_mysql_acct_check_tables(m
 				  "auto_increment=1001")
 	    == SLURM_ERROR)
 		return SLURM_ERROR;
-	else {
-		if ((rc = as_mysql_convert_get_bad_tres(mysql_conn)) !=
-		    SLURM_SUCCESS) {
-			error("issue getting bad tres");
-			return rc;
-		} else if (backup_dbd) {
-			/*
-			 * We do not want to create/check the database if we are
-			 * the backup (see Bug 3827). This is only handled on
-			 * the primary.
-			 */
-			return rc;
-		}
+
+	/* This has to run on both backup as well as primary */
+	if ((rc = as_mysql_convert_get_bad_tres(mysql_conn)) != SLURM_SUCCESS) {
+		error("issue getting bad tres");
+		return rc;
+	} else if (!backup_dbd) {
 		/* We always want CPU to be the first one, so create
 		   it now.  We also add MEM here, the others tres
 		   are site specific and could vary.  None but CPU
@@ -917,7 +910,13 @@ static int _as_mysql_acct_check_tables(m
 		 * We do not want to create/check the database if we are the
 		 * backup (see Bug 3827). This is only handled on the primary.
 		 */
+
 		slurm_mutex_unlock(&as_mysql_cluster_list_lock);
+
+		/* We do want to set the QOS count though. */
+		if (rc == SLURM_SUCCESS)
+			rc = _set_qos_cnt(mysql_conn);
+
 		return rc;
 	}
 
